### DISCUSSION TITLE
Has anyone ever managed to extend the document model?

### DISCUSSION THREAD

**Discussion by mrpdean** [pimcore/pimcore#11704] [Q&A / Support]
I've been bashing my head against this for a while now and there is so little documentation available for it.

I'm wondering if anyone has every been able to extend the Document model to add additional fields?

Pimcore's Data Object functionality is absolutely fantastic and so it seems odd to me that there doesn't seem to be a way to extend Documents in order to add additional fields and functionality.

The few responses I've seen to similar questions tend to steer people towards using a combination of Data Objects, associated with Documents, but surely there is a neater, more programmatic way of doing it?

**Comment by dpfaffenbauer** [comment]
There is no need in extending documents. You can use properties for example if you want to add new "fields"

**Reply by mrpdean** [reply]
That's true however the data types for properties are limited.  For example, there is no type to display a date picker, similar to: [https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Date_Types.html](https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Date_Types.html)
